src/common/mod.rs

```diff
@@ -180,6 +180,38 @@ where
             };
         }
     }
+
+    pub(crate) fn poll_fill_buf(mut self, cx: &mut Context<'_>) -> Poll<io::Result<&'a [u8]>>
```

Comment:
Should implement `AsyncBufRead` for `common::Stream` and test it in `common/test_stream.rs`?